presto-hive/src/test/java/com/facebook/presto/hive/TestPartitionUpdate.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.PartitionUpdate.UpdateMode;
+import com.google.common.collect.ImmutableList;
+import io.airlift.json.JsonCodec;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.Test;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static org.testng.Assert.assertEquals;
+
+public class TestPartitionUpdate
+{
+    private static final JsonCodec<PartitionUpdate> CODEC = jsonCodec(PartitionUpdate.class);
+
+    @Test
+    public void testRoundTrip()
+    {
+        PartitionUpdate expected = new PartitionUpdate(
+                "test",
+                UpdateMode.APPEND,
+                "/writePath",
+                "/targetPath",
+                ImmutableList.of("file1", "file3"),
+                123,
+                456,
+                789);
+
+        PartitionUpdate actual = CODEC.fromJson(CODEC.toJson(expected));
+
+        assertEquals(actual.getName(), "test");
```

Comment:
instead of hardcoding the literals twice maybe just change these to call the getter on the expected... so `assertEquals(actual.getName(), expected.getName());`

Reply:
I've actually given the opposite advice recently. This is more verbose, but makes sure we didn't screw up the constructor call ordering or assignment, avoids copy/paste error in a line of assertion calls, and is arguably easier to read. (this would be less of a concern if we had other tests to cover these aspects)